### PR TITLE
[PORT] Crysis suit from hippie haha.

### PR DESCRIPTION
**Pull request by Moccha-Bee**
## About The Pull Request
Adds the crysis nanosuit from hippie that costs 20TC and acts like a ninjasuit with more options but less power. As well as the emag option that increases the drain in power of the suit when  in certain modes but increases their funcionality.

From: https://github.com/HippieStation/HippieStation/pull/11995
From:https://github.com/HippieStation/HippieStation/pull/7540

## Why It's Good For The Game
Syndie ninja time.

## Changelog
:cl:
add: Added crysis nanosuit
/:cl:

### CODE REVIEW

**Comment by Moccha-Bee** [comment]
I did a stinky  going to fix it real quick

File: code/game/objects/items/hfblade.dm [hunk 1]
```diff
@@ -0,0 +1,142 @@
+// Chaplains begone!
+/obj/item/nullrod/scythe/vibro
+	name = "high fronquency bladd"
+
+/obj/item/melee/hfblade
```
Comment (MarkSuckerberg) [inline code/game/objects/items/hfblade.dm]: You have the HF blade in this PR.

File: beestation.dme [hunk 2]
```diff
@@ -1633,6 +1633,7 @@
 #include "code\modules\clothing\spacesuits\chronosuit.dm"
 #include "code\modules\clothing\spacesuits\hardsuit.dm"
 #include "code\modules\clothing\spacesuits\miscellaneous.dm"
+#include "code\modules\clothing\spacesuits\nanosuit.dm"
```
Comment (MarkSuckerberg) [inline beestation.dme]: Here too

File: code/modules/uplink/uplink_items.dm [hunk 3]
```diff
@@ -319,6 +319,20 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	surplus = 30
 	include_modes = list(/datum/game_mode/nuclear)
 
+/datum/uplink_item/dangerous/vibroblade
```
Comment (MarkSuckerberg) [inline code/modules/uplink/uplink_items.dm]: And here

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]
Uhhhh fix

File: code/modules/uplink/uplink_items.dm (hunk 3 above)
Comment (Moccha-Bee) [reply code/modules/uplink/uplink_items.dm]: already removed it

File: code/game/objects/items/hfblade.dm (hunk 1 above)
Comment (Moccha-Bee) [reply code/game/objects/items/hfblade.dm]: removed it as well.

**Comment by Moccha-Bee** [comment]
So i can't remove the HF blade sound files and sprite files oops.

**Comment by Ruediger4** [comment]
oh no

File: code/modules/clothing/spacesuits/nanosuit.dm [hunk 4]
```diff
@@ -0,0 +1,1269 @@
...
+	desc = "Ashes to ashes."
+	icon_state = "explosive"
+	resistance_flags = INDESTRUCTIBLE | FIRE_PROOF | ACID_PROOF | FREEZE_PROOF
+
+/obj/item/implant/explosive/disintegrate/activate(cause)
+	if(!cause || !imp_in || cause == "emp" || active)
+		return FALSE
+	if(cause == "action_button" && !popup)
+		popup = TRUE
+		var/response = alert(imp_in, "Are you sure you want to activate your [name]? This will cause you to disintergrate!", "[name] Confirmation", "Yes", "No")
```
Comment (MarkSuckerberg) [inline code/modules/clothing/spacesuits/nanosuit.dm]: Disintergrate

**Review by MarkSuckerberg — COMMENTED** [review]
Also you seem to have left in the sfx from the HF blade

**Comment by MarkSuckerberg** [comment]
And you need to delete the sprites and sounds or there will likeley be a conflict if one gets merged

**Review by Moccha-Bee — COMMENTED** [review]
> And you need to delete the sprites and sounds or there will likeley be a conflict if one gets merged

I can't delete them for some reason...

File: code/modules/clothing/spacesuits/nanosuit.dm (hunk 4 above)
Comment (Moccha-Bee) [reply code/modules/clothing/spacesuits/nanosuit.dm]: There is a suicide button like the one nuke ops get but this one makes you desintegrate.

**Comment by MarkSuckerberg** [comment]
Just, like, delete them locally and you should be able to commit the removals and push them

File: code/modules/clothing/spacesuits/nanosuit.dm (hunk 4 above)
Comment (MarkSuckerberg) [reply code/modules/clothing/spacesuits/nanosuit.dm]: No, I meant it's mispelled in this line

Comment (Moccha-Bee) [reply code/modules/clothing/spacesuits/nanosuit.dm]: oh yeah it is will fix it reaal soon.

**Comment by Moccha-Bee** [comment]
> Just, like, delete them locally and you should be able to commit the removals and push them

Thanks, Its done

**Comment by SomeAngryMiner** [comment]
am i seeing this incorrectly, or does the nanosuit dont exist at all, outside of the uplink and icons?

**Comment by SomeAngryMiner** [comment]
either that or its a ninja suit child

**Comment by Moccha-Bee** [comment]
> either that or its a ninja suit child

well it worked just fine on my private server and probably is a ninja suit child considering how it works the same way.

**Comment by MarkSuckerberg** [comment]
> am i seeing this incorrectly, or does the nanosuit dont exist at all, outside of the uplink and icons?

See nanosuit.dm

File: code/modules/clothing/spacesuits/nanosuit.dm [hunk 5]
```diff
@@ -0,0 +1,1269 @@
...
+	..()
+	//STOP_PROCESSING(SSfastprocess, src)
+	var/datum/atom_hud/secsensor = GLOB.huds[DATA_HUD_SECURITY_ADVANCED]
+	var/datum/atom_hud/diagsensor = GLOB.huds[DATA_HUD_DIAGNOSTIC_BASIC ]
+	secsensor.remove_hud_from(Wearer)
+	diagsensor.remove_hud_from(Wearer)
+	if(help_verb && Wearer)
+		Wearer.verbs -= help_verb
+
+/obj/item/clothing/suit/space/hardsuit/nano/proc/bootSequence()
```
Comment (qwertyquerty) [inline code/modules/clothing/spacesuits/nanosuit.dm:746]: I do not believe this is up to current coding standards

**Comment by hackerjsimon** [comment]
Code a Lil chunky like qwerty pointed out

also definitely not 20 tc if its available to traitors 26-27 are good prices for antag roles with access.

File: code/modules/uplink/uplink_items.dm [hunk 6]
```diff
@@ -324,6 +324,7 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	desc = "Become a posthuman warrior. The items cannot be taken off once you wear them."
 	item = /obj/item/storage/box/syndie_kit/nanosuit
 	cost = 20
```
Comment (hackerjsimon) [inline code/modules/uplink/uplink_items.dm]: change the price to 20

Comment (Moccha-Bee) [reply code/modules/uplink/uplink_items.dm]: Its already 20

**Comment by Moccha-Bee** [comment]
Update: After community feedback i increased the price for the nanosuit so that it isn't spammed in every single traitor round, it now has a minimum pop recquirement, can't be brought by ops, and now costs 25TC

File: code/modules/clothing/spacesuits/nanosuit.dm (hunk 5 above)
Comment (Moccha-Bee) [reply code/modules/clothing/spacesuits/nanosuit.dm:746]: "sleep"

Comment (Moccha-Bee) [reply code/modules/clothing/spacesuits/nanosuit.dm:746]: ?

**Review by ike709 — CHANGES_REQUESTED** [review]
Not a fan. There's serious code quality issues, to the point where almost all of this needs to be rewritten.

This is about as bad as old ninjacode, for less content.

**Comment by ike709** [comment]
This PR is significantly below our standards and would require a rather large rewrite. Therefore, I'm closing it.

**Pull request closed without merging**